Sources/AsyncHTTPClient/ConnectionsState.swift

```diff
@@ -229,7 +229,6 @@ extension HTTP1ConnectionProvider {
                 self.openedConnectionsCount -= 1
                 return self.processNextWaiter()
             case .closed:
-                assertionFailure("should not happen")
```

Comment:
maybe we should add a comment when this can happen?

Reply:
good idea, done